### PR TITLE
Feature cluster force layout

### PR DESCRIPTION
**Pull request by samshara**

### CODE REVIEW

File: components/Visualization/ClusterForceLayout/index.js [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+class ClusterForceLayout extends PureComponent {
+    static propTypes = propTypes;
+    static defaultProps = defaultProps;
+
+    componentDidMount() {
+        this.drawChart();
+    }
+    componentWillReceiveProps(nextProps) {
+        if (nextProps.data !== this.props.data) {
+            this.updateData(nextProps);
```
Comment (tnagorra) [inline components/Visualization/ClusterForceLayout/index.js]: there is no method updateData

File: components/Visualization/ClusterForceLayout/index.js [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+            .style('top', `${event.pageY - 50}px`)
+            .style('left', `${event.pageX - 30}px`);
+    }
+
+    onMouseOut = () => {
+        select(this.tooltip)
+            .style('display', 'none');
+    }
+
+    getMaxNode = (data, valueSelector) => data.reduce((previous, current) => (
```
Comment (tnagorra) [inline components/Visualization/ClusterForceLayout/index.js]: getMaxNode not used

File: components/Visualization/ClusterForceLayout/index.js [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+} from 'd3-force';
+import { forceAttract } from 'd3-force-attract/index';
+import { forceCluster } from 'd3-force-cluster/index';
+
+import Responsive from '../../General/Responsive';
+import Float from '../../View/Float';
+
+import {
+    groupList,
+    getStanadardFilename,
```
Comment (tnagorra) [inline components/Visualization/ClusterForceLayout/index.js]: Stanadard is written as standard

File: components/Visualization/ClusterForceLayout/index.js [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+        d.fy = event.y; // eslint-disable-line no-param-reassign
+    }
+
+    dragended = (d) => {
+        if (!event.active) this.simulation.alphaTarget(0);
+        d.fx = null; // eslint-disable-line no-param-reassign
+        d.fy = null; // eslint-disable-line no-param-reassign
+    }
+
+    save = () => {
```
Comment (tnagorra) [inline components/Visualization/ClusterForceLayout/index.js:148]: save should not be called using ref

File: components/Visualization/ClusterForceLayout/index.js [hunk 5]
```diff
@@ -0,0 +1,302 @@
...
+
+        const {
+            top,
+            right,
+            bottom,
+            left,
+        } = margins;
+        if (!boundingClientRect.width) return;
+
+        this.width = width - left - right;
```
Comment (tnagorra) [inline components/Visualization/ClusterForceLayout/index.js]: can width and height not be set on this

File: components/Visualization/ClusterForceLayout/index.js [hunk 6]
```diff
@@ -0,0 +1,302 @@
...
+            const element = {
+                id: idSelector(node),
+                radius,
+                group,
+                x: (Math.cos((group / noOfClusters) * 2 * Math.PI) * 200) +
+                      (width / 2) + Math.random(),
+                y: (Math.sin((group / noOfClusters) * 2 * Math.PI) * 200) +
+                      (height / 2) + Math.random(),
+            };
+            if (!clusters.find(cluster => cluster.group === group) ||
```
Comment (tnagorra) [inline components/Visualization/ClusterForceLayout/index.js]: Double computation of clusters.find

File: components/Visualization/ClusterForceLayout/index.js (hunk 4 above)
Comment (samshara) [reply components/Visualization/ClusterForceLayout/index.js:148]: the save function is not called with ref.

Comment (tnagorra) [reply components/Visualization/ClusterForceLayout/index.js:148]: nice

**Pull request merged**